fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionUtil.java

```diff
@@ -509,4 +516,52 @@ public static Type getPartitionColumnType(MTMVRelatedTableIf relatedTable, Strin
         }
         throw new AnalysisException("can not getPartitionColumnType by:" + col);
     }
+
+    public static List<String> getPartitionsByRange(MTMV mtmv, MTMVRefreshPartitionRange range)
+            throws AnalysisException {
+        List<String> res = Lists.newArrayList();
+        Collection<Partition> partitions = mtmv.getPartitions();
+        for (Partition partition : partitions) {
+            PartitionItem item = mtmv.getPartitionInfo().getItem(partition.getId());
+            if (ifPartitionItemInRange(item, range)) {
+                res.add(partition.getName());
+            }
+        }
+        return res;
+    }
+
+    public static boolean ifPartitionItemInRange(PartitionItem item, MTMVRefreshPartitionRange refreshPartitionRange)
+            throws AnalysisException {
+        Range<DateV2Literal> range = refreshPartitionRange.getRange();
+        if (item instanceof ListPartitionItem) {
+            ListPartitionItem listPartitionItem = (ListPartitionItem) item;
+            List<PartitionKey> keys = listPartitionItem.getItems();
+            for (PartitionKey key : keys) {
+                if (range.contains(transferPartitionKeyToDate(key))) {
+                    return true;
+                }
+            }
+        } else if (item instanceof RangePartitionItem) {
+            RangePartitionItem rangePartitionItem = (RangePartitionItem) item;
+            Range<PartitionKey> items = rangePartitionItem.getItems();
+            DateV2Literal lowerDate = transferPartitionKeyToDate(items.lowerEndpoint());
+            DateV2Literal upperDate = transferPartitionKeyToDate(items.upperEndpoint());
+            Range<DateV2Literal> partitionRange = Range.closedOpen(lowerDate, upperDate);
+            return range.isConnected(partitionRange) && !range.intersection(partitionRange).isEmpty();
+        }
+        return false;
+    }
+
+    private static DateV2Literal transferPartitionKeyToDate(PartitionKey partitionKey) throws AnalysisException {
```

Comment:
if mv is partitoned by hour, maybe we should support datatime? it may be general

Reply:
If there is a need, I will implement it in another PR